src/Compilers/CSharp/Portable/Symbols/SymbolExtensions.cs

```diff
@@ -832,5 +819,16 @@ internal static int GetOverloadResolutionPriority(this Symbol symbol)
             Debug.Assert(symbol is MethodSymbol or PropertySymbol);
             return symbol is MethodSymbol method ? method.OverloadResolutionPriority : ((PropertySymbol)symbol).OverloadResolutionPriority;
         }
+
+        internal static bool IsExtensionParameter(this ParameterSymbol parameter)
+        {
+            return parameter.ContainingSymbol is TypeSymbol { IsExtension: true };
+        }
+
+        internal static bool IsExtensionParameterImplementation(this ParameterSymbol parameter)
+        {
+            return parameter.ContainingSymbol is SourceExtensionImplementationMethodSymbol implementationMethod
```

Comment:
>parameter [](http://example.com/codeflow?start=19&length=9)

It looks like we are making an assumption that this method is always called on a definition. Consider asserting the fact.